rotkehlchen/chain/ethereum/modules/yearn/vaultsv2.py

```diff
@@ -215,15 +221,23 @@ def get_vaults_history(
                     to_block=to_block,
                 )
                 continue
+            self.database.add_yearn_vaults_events(address, events)
+            addresses_to_events[address] = db_events + events
 
+        for address in addresses:
+            if address in addresses_to_events:
+                all_events = addresses_to_events[address]
+            else:
+                all_events = self.database.get_yearn_vaults_v2_events(
+                    address=address,
+                    from_block=from_block,
+                    to_block=to_block,
+                )
```

Comment:
Would't this be simpler like so:

```suggestion
            all_events = self.database.get_yearn_vaults_v2_events(
                address=address,
                from_block=from_block,
                to_block=to_block,
            )
```

Don't take my word for it, but if I read the code correctly, at this point you have already input all new events in the DB so you can just query it from there.

This way you can also delete all the logic of combinng existing DB events with what is returned from the subgraph that you have above. Does that make sense?

Reply:
Is true, I could avoid this